File: songify-frontend/src/services/UserService.js
import axios from 'axios'

const config = require('../config.json');

class UserService{

    getUsers(){
        return axios.get(config.REST_API_URL + "users");
    }

    getUserById(id){
        return axios.get(config.REST_API_URL + "users/" + id)
    }

    updateUser(user, id){
        return axios.put(config.REST_API_URL + "users/" + id, user)
    }

    deleteUser(id){
        return axios.delete(config.REST_API_URL + "users/" + id)
    }
}

export default new UserService();

File: songify-backend/src/main/java/com/songify/api/service/UserService.java
package com.songify.api.service;

import com.songify.api.model.dto.AcceptFriendRequestRequest;
import com.songify.api.model.dto.UserDto;
import com.songify.api.exceptions.UserNotFoundException;
import com.songify.api.model.User;

import java.util.List;

public interface UserService {

    List<User> getAllUsers();

    User getUserById(Long id) throws UserNotFoundException;

    User getUserByUsername(String username);

    User getUserByUsernameAndPassword(String username, String password);

    User addUser(UserDto dto);

    User updateUser(Long userId, UserDto dto);

    String deleteUser(Long id);

    User save(User user);
}